packages/shared-scripts/src/calculate-file-sizes.js

```diff
@@ -0,0 +1,139 @@
+import fs from "fs/promises";
+import path from "path";
+import json2md from "json2md";
+
+// Get directory and flags from command line arguments
+const args = process.argv.slice(2);
+const relativeDir = args[0];
+
+const isBefore = args.includes("--before");
+const isAfter = args.includes("--after");
+
+if (!relativeDir) {
+  console.log("Please provide a directory path as an argument.");
+  process.exit(1);
+}
+
+if (!isBefore && !isAfter) {
+  console.log("Please specify either --before or --after flag.");
+  process.exit(1);
+}
+
+// Get the working directory from the environment variable
+const scriptDir = process.env.WORKING_DIR ?? process.cwd();
+
+const dir = path.resolve(scriptDir, relativeDir);
+const jsonFilePath = path.join(scriptDir, "file-sizes.json");
+
+const calculateFileSizes = async () => {
+  try {
+    // Check if the directory exists
+    await fs.access(dir);
+
+    const files = await fs.readdir(dir);
+
+    let fileSizes = { beforeOptimization: {}, afterOptimization: {} };
+
+    // Check if the JSON file already exists and load data
+    try {
+      const data = await fs.readFile(jsonFilePath, "utf-8");
+      fileSizes = JSON.parse(data);
+    } catch {
+      // If file doesn't exist, initialize fileSizes as default
+    }
+
+    // If the --after flag is used, ensure beforeOptimization data exists
+    if (isAfter && Object.keys(fileSizes.beforeOptimization).length === 0) {
+      console.log(
+        "No data found before optimization. Please run the script with --before first."
+      );
+      process.exit(1);
+    }
+
+    // Filter .wasm files and calculate sizes
+    const wasmFiles = files.filter((file) => path.extname(file) === ".wasm");
+
+    // Wait for all file size calculations to complete
+    await Promise.all(
+      wasmFiles.map(async (file) => {
+        const filePath = path.join(dir, file);
+        const stats = await fs.stat(filePath);
+
+        const fileSizeInKB = (stats.size / 1024).toFixed(2);
+
+        // Add file size to the appropriate section
+        if (isBefore) {
+          fileSizes.beforeOptimization[file] = `${fileSizeInKB} KB`;
+        } else if (isAfter) {
+          fileSizes.afterOptimization[file] = `${fileSizeInKB} KB`;
+        }
+      })
+    );
+
+    // Write the result to the JSON file
+    await fs.writeFile(jsonFilePath, JSON.stringify(fileSizes, null, 2));
+
+    console.log(`File sizes saved to ${jsonFilePath}`);
+
+    const updatedData = await fs.readFile(jsonFilePath, "utf-8");
+
+    const updatedFileSizes = JSON.parse(updatedData);
+
+    if (
+      Object.keys(updatedFileSizes.beforeOptimization).length &&
+      Object.keys(updatedFileSizes.afterOptimization).length
+    ) {
+      // Generate Markdown file
+      generateMarkdown(scriptDir, fileSizes);
```

Comment:
While you do catch errors in `calculateFileSizes`, some of the operations (e.g., generateMarkdown) don't have explicit error handling. It’s a good idea to wrap generateMarkdown in a try-catch block in case there are issues during markdown generation or file operations.